Utils/CommonDriver.cs

```diff
@@ -0,0 +1,150 @@
+﻿using AventStack.ExtentReports;
+using AventStack.ExtentReports.Reporter;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using TechTalk.SpecFlow;
+using System;
+using System.IO;
+using NUnit.Framework;
+using MarsAdvancedTaskPart2.Pages;
+using MarsAdvancedTaskPart2.Pages.Components.Profilepage;
+using MarsAdvancedTaskPart2.Pages.Components;
+using MarsAdvancedTaskPart2.TestModel;
+
+namespace MarsAdvancedTaskPart2.Utils
+{
+
+    public class CommonDriver
+    {
+        CommonDriver commonDriverObj;
+        LoginPage loginPageObj;
+        ProfileTabEducation educationObj;
+        ProfileTabCertifications certificateObj;
+        ManageListings manageListingsObj;
+
+        public static IWebDriver driver;
+        public static ExtentReports extent;
+        public static ExtentTest test;
+        public static List<string> EducationDataToCleanUp { get; set; } = new List<string>();
+        public static List<string> CertificateDataToCleanUp { get; set; } = new List<string>();
+        public static List<string> ManageListingToCleanUp { get; set; } = new List<string>();
+
+
+
+        [BeforeScenario]
+        public void BeforeScenario()
+        {
+            var sparkReporter = new ExtentSparkReporter(@"D:\MarsAdvancedTaskPart2\Reports\extentReport.html");
+            extent = new ExtentReports();
+            extent.AttachReporter(sparkReporter);
+
+            driver = new ChromeDriver();
+            driver.Manage().Window.Maximize();
+            driver.Navigate().GoToUrl("http://localhost:5000/Home");
+
+            loginPageObj = new LoginPage();
+            educationObj = new ProfileTabEducation();
+            certificateObj = new ProfileTabCertifications();
+            manageListingsObj = new ManageListings();
+
+            // Create test entry in the report
+            var testName = TestContext.CurrentContext.Test.Name;
+            test = extent.CreateTest(testName);
+           
+        }
+
+        public static void TakeScreenshotWithPngFormat()
+        {
+            try
+            {
+                Screenshot screenshot = ((ITakesScreenshot)driver).GetScreenshot();
+                screenshot.SaveAsFile(@"D:\MarsAdvancedTaskPart2\ScreenShot\Screenshot." + System.Drawing.Imaging.ImageFormat.Png);
+
+
+                // Add screenshot to ExtentReports
+
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"An error occurred while taking screenshot: {e.Message}");
+            }
+        }
+
+        [AfterScenario]
+        public void AfterScenario()
+        {
+            try
```

Comment:
good to see clean up of data for each test but what about clean up of all data at start (before any test run) you should clean up all data